### PR TITLE
Fix datarace

### PR DESCRIPTION
**Pull request by b00ris**
```
==================
WARNING: DATA RACE
Write at 0x00c01d871670 by goroutine 817:
  github.com/clearmatics/autonity/p2p.(*Server).UpdateWhitelist()
      /Users/boris/go/src/github.com/clearmatics/autonity/p2p/server.go:406 +0x772
  github.com/clearmatics/autonity/eth.(*Ethereum).glienickeEventLoop()
      /Users/boris/go/src/github.com/clearmatics/autonity/eth/backend.go:635 +0x71b

Previous read at 0x00c01d871670 by goroutine 398:
  github.com/clearmatics/autonity/consensus/test.runTest.func4()
      /Users/boris/go/src/github.com/clearmatics/autonity/consensus/test/tendermint_test.go:1091 +0xe4
  github.com/clearmatics/autonity/vendor/golang.org/x/sync/errgroup.(*Group).Go.func1()
      /Users/boris/go/src/github.com/clearmatics/autonity/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x71
```

### CODE REVIEW

File: consensus/test/tendermint_test.go [hunk 1]
```diff
@@ -1086,11 +1086,13 @@ func runTest(t *testing.T, test *testCase) {
 		i := i
 
 		wg.Go(func() error {
+			validator.node.Server().Lock.Lock()
```
Comment (JekaMas) [inline consensus/test/tendermint_test.go]: I think it's removed by Aneeque.

**Comment by JekaMas** [comment]
https://github.com/clearmatics/autonity/pull/296/files#diff-452fb4e9f04a68a05a25df13b67e6842L878 Could you have a look at the @aneequesafdar PR ?

**Comment by aneequesafdar** [comment]
So there is a data race on trusted nodes and static nodes we can do it either way but I was also getting a deadlock once I removed (https://github.com/clearmatics/autonity/pull/296/files#diff-452fb4e9f04a68a05a25df13b67e6842L878). So maybe using locks might be better?

**Comment by aneequesafdar** [comment]
Since this was only logging information and while running the tests the only Error messages were printed so I thought it might be best to remove the log, however, it up to you. I didn't want to add getters and setters for trusted and static nodes, I didn't know there was a lock present for the server so it might be best to use it?

**Comment by b00ris** [comment]
I added lock because i thought that it could be useful if we want to write tests on network permissioning feature. We'll need a lock for checking static peers state.

**Comment by aneequesafdar** [comment]
If you think its the best way seems good to me

**Comment by JekaMas** [comment]
@b00ris approved

**Pull request closed without merging**